src/pybind/mgr/dashboard/controllers/multi_cluster.py

```diff
@@ -375,25 +431,116 @@ def check_token_status(self, clustersTokenMap=None):
-            valid_ip_url = bool(re.match(pattern, prometheus_url))
-            if not valid_ip_url:
-                parsed_url = urlparse(prometheus_url)
-                hostname = parsed_url.hostname
-                orch = OrchClient.instance()
-                inventory_hosts = [host.to_json() for host in orch.hosts.list()]
+            if '.ceph-dashboard' in prometheus_url:
+                prometheus_url = prometheus_url.replace('.ceph-dashboard', '')
+            parsed_url = urlparse(prometheus_url)
+            hostname = parsed_url.hostname
+            try:
+                # Check if the hostname is already an IP address
+                ipaddress.ip_address(hostname)
+                valid_ip_url = True
+            except ValueError:
+                valid_ip_url = False
+
+            orch = OrchClient.instance()
+            inventory_hosts = (
+                [host.to_json() for host in orch.hosts.list()]
+                if not valid_ip_url
+                else []
+            )
+
+            def find_node_ip():
                 for host in inventory_hosts:
-                    if host['hostname'] == hostname or host['hostname'] in hostname:
-                        node_ip = host['addr']
-                prometheus_url = prometheus_url.replace(hostname, node_ip)
+                    if host['hostname'] == hostname or hostname in host['hostname']:
+                        return host['addr']
+                return None
+
+            if mgmt_gw_enabled:
+                if not valid_ip_url:
+                    node_ip = find_node_ip()
+                    if node_ip:
+                        prometheus_url = node_ip
+                else:
+                    prometheus_url = hostname
+            else:
+                if not valid_ip_url:
+                    node_ip = find_node_ip()
+                    if node_ip:
+                        prometheus_url = prometheus_url.replace(hostname, node_ip)
         return prometheus_url
 
+    def find_prometheus_credentials(self, multicluster_config: Dict[str, Any],
+                                    url: str) -> Optional[Dict[str, Any]]:
+        for _, clusters in multicluster_config['config'].items():
+            for cluster in clusters:
+                prometheus_url = cluster.get('prometheus_url')
+                if prometheus_url:
+                    clean_prometheus_url = (
+                        prometheus_url.replace("https://", "").replace("http://", "")
+                    )  # since target URLs are without scheme
+
+                    if clean_prometheus_url == url:
+                        return cluster.get('prometheus_access_info')
+        return None
+
+    def get_cluster_credentials(self, targets: List[str]) -> Dict[str, Any]:
+        clusters_credentials = {}
+        multi_cluster_config = self.load_multi_cluster_config()
+        try:
+            if multi_cluster_config:
+                for url in targets:
```

Comment:
we can return earlier if `multi_cluster_config` is empty by:

```
if multi_cluster_config:
   return {}
```
This way we avoid the extra and unnecessary indentation in the try-except block

Reply:
Updated